core/src/main/java/org/fao/geonet/kernel/search/spatial/SpatialIndexWriter.java

```diff
@@ -506,18 +517,107 @@ public static Name findIdColumn(FeatureSource<SimpleFeatureType, SimpleFeature>
         return null;
     }
 
+    /**
+     * Produces a multipolygon that covers the provided geometry.
+     *
+     * @param geometry
+     * @return covering multipolygon
+     */
     public static MultiPolygon toMultiPolygon(Geometry geometry) {
-        if (geometry instanceof Polygon) {
-            Polygon polygon = (Polygon) geometry;
-
-            return geometry.getFactory().createMultiPolygon(
-                new Polygon[]{polygon});
-        } else if (geometry instanceof MultiPolygon) {
+        if (geometry == null) return null;
+        if (geometry instanceof MultiPolygon) {
             return (MultiPolygon) geometry;
         }
-        String message = geometry.getClass() + " cannot be converted to a polygon. Check metadata";
-        Log.error(Geonet.INDEX_ENGINE, message);
-        throw new IllegalArgumentException(message);
+        final double DISTANCE = 0.01;
```

Comment:
This ends up being around 1km "approximate metric equivalent for 0.01 degree". The real goal here is to have a non empty bounding box so math does not end up failing us. Around 1 km is civic scale and should be visible if zoomed in against OSM.